### PR TITLE
Ignore cost model failures

### PR DESCRIPTION
**Pull request by arhimondr**
Currently stats and costs are pre-computed for every query.
There are still some bugs in stats calculator that are being
triggered in very rare cases that are incredibly hard to catch.
Given that the stats calculator has to be run for every query, it
is better to be fail safe. As a failure in stats calculation must
not result in failing query.

### CODE REVIEW

**Comment by arhimondr** [comment]
#11856 addresses the failures we discovered and introduces additional improvements. While #11856 requires intensive testing and more long lasting discussions, this patch tries to mitigate the failures without completely disabling cost calculations.

Given the rarity of the failures this patch may also be beneficial for a long term as well.

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 1]
```diff
@@ -525,7 +526,12 @@ public SystemSessionProperties(
                         null,
                         false,
                         value -> min(taskManagerConfig.getMaxDriversPerTask(), validateNullablePositiveIntegerValue(value, MAX_DRIVERS_PER_TASK)),
-                        object -> object));
+                        object -> object),
+                booleanProperty(
+                        IGNORE_STATS_CALCULATOR_FAILURES,
+                        "Experimental: Ignore statistics calculator failures rather than failing",
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: I don't think it should be marked experimental.  It's a safety mechanism.

Comment (mbasmanova) [reply presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java]: +1

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java [hunk 2]
```diff
@@ -871,4 +872,17 @@ public FeaturesConfig setLegacyUnnestArrayRows(boolean legacyUnnestArrayRows)
         this.legacyUnnestArrayRows = legacyUnnestArrayRows;
         return this;
     }
+
+    public boolean isIgnoreStatsCalculatorFailures()
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: Please put the methods in the same order as the properties (I'd expect this to be below `enableStatsCalculator`)

**Comment by electrum** [comment]
Let's rename the "Revert ..." commit to be "Enable stats calculator by default", since that's more straightforward and isn't a double negative. It's fine to leave the revert reference in the description.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java (hunk 2 above)
Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: All the methods has to be reordered in that class. I moved this very method after `enableStatsCalculator`, but methods before `enableStatsCalculator` are not in the correct order.

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestFeaturesConfig.java [hunk 3]
```diff
@@ -104,7 +104,8 @@ public void testDefaults()
                 .setMultimapAggGroupImplementation(MultimapAggGroupImplementation.NEW)
                 .setDistributedSortEnabled(true)
                 .setMaxGroupingSets(2048)
-                .setLegacyUnnestArrayRows(false));
+                .setLegacyUnnestArrayRows(false)
+                .setIgnoreStatsCalculatorFailures(true));
```
[the remark this answers is not in the document]
Comment (arhimondr) [reply presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestFeaturesConfig.java]: Moved. The problem is that `setenableStatsCalculator` was also order after the preceeding method. And the preceeding methods (`setIterativeOptimizerEnabled`, `setIterativeOptimizerTimeout`) aren't ordered correctly. We need to rearrenge the code in FC and TFC. I just don't feel like doing it right now. That's a lot of manual labor =))

**Comment by arhimondr** [comment]
I decided to go with a default value of the ignore property set to `false`. We shouldn't be ignoring by default in all of the possible tests and so on. Especially after the #11856 fixes all the problems we have discovered so far, this property must be set to true only in production environments if any issues are discovered.

**Pull request merged**